src/baldr/incident_singleton.h

```diff
@@ -249,25 +283,24 @@ struct incident_singleton_t {
       auto wait = 0;
       if (latency > max_loading_latency) {
         LOG_WARN("Incident watcher is not meeting max loading latency requirement");
-      } // this round finished fast enough, changelog mode is cheap so wait only a second
+      } // this round finished fast enough wait the remainder of the time specified
       else {
-        wait = changelog ? 1 : max_loading_latency - latency;
+        wait = max_loading_latency - latency;
       }
-      LOG_INFO("Incident watcher loaded " + std::to_string(update_count) + " tiles in " +
+      LOG_INFO("Incident watcher updated " + std::to_string(update_count) + " tiles in " +
                std::to_string(latency) + " seconds");
 
       // signal to the constructor that we completed our first batch
-      if (first_run) {
+      if (run_count++ == 0) {
         LOG_INFO("Incident watcher initialized");
-        first_run = false;
         state->signal.notify_one();
       }
 
       // wait just a little before we check again
       std::this_thread::sleep_for(std::chrono::seconds(wait));
-    } while (true);
+    } while (!interrupt || !interrupt(run_count));
```

Comment:
if you set an interrupt and if the interrupt returns true on this interation, then we bail. needed for unit testing only